pkg/controllers/v1alpha2/mpi_job_controller.go

```diff
@@ -440,12 +439,15 @@ func (c *MPIJobController) syncHandler(key string) error {
 		return err
 	}
 	// We're done if the launcher either succeeded or failed.
-	done := launcher != nil && (launcher.Status.Succeeded == 1 || launcher.Status.Failed == 1)
+	done := launcher != nil && isJobFinished(launcher)
 
-	workerSpec := mpiJob.Spec.MPIReplicaSpecs[kubeflow.MPIReplicaTypeWorker]
-	workerReplicas := *workerSpec.Replicas
+	// If MPIJob have done, workerReplicas will set to default 0.
+	var workerReplicas int32
 
 	if !done {
+		workerSpec := mpiJob.Spec.MPIReplicaSpecs[kubeflow.MPIReplicaTypeWorker]
+		workerReplicas = *workerSpec.Replicas
```

Comment:
only set workerReplicas when launcher haven't done, If launcher have done, left  it workerReplicas as default 0

Reply:
Why setting to 0? Once MPIJob is finished, all worker pods should remain undeleted there so this number should not change either, right?

Reply:
In version v1alpha1, we just set it to 0. When the launcher has completed, What is the use of the worker :)? 
And in the product env, if don't delete this worker pod, that will be always spend the user's expense.
 How about add a field in the MPIjob, then user can decide whether to delete？

Reply:
Like the [CleanUpPolicy](https://github.com/kubeflow/tf-operator/blob/master/pkg/apis/tensorflow/v1/types.go#L56) in tf-operator

Reply:
Yes, this needs to be customized. We can replace https://github.com/kubeflow/mpi-operator/blob/master/pkg/apis/kubeflow/v1alpha2/common_types.go with https://github.com/kubeflow/common/blob/master/job_controller/api/v1/types.go, which exposes the cleanup policy.

Reply:
The CleanPodPolicy of job will be Running/All/None, but CleanPodPolicy of statefulset will be whether to set the replica to 0... maybe can't use the common api directly
Add a commit to add a bool field CleanPodPolicy, how about it?

Reply:
Running/All set to 0, None/"" don't set to 0?

Reply:
I am not sure if it's necessary to set it to 0. We can just reuse CleanPodPolicy from kubeflow/common to decide whether to delete the pod groups.

Reply:
Have add the CleanPodPolicy from common_type.go, now user can customize the CleanPodPolicy then decide whether delete the worker.